### ISSUE TITLE
Drawer/Modal A11Y Issue, trap focus tabIndex=1

### ISSUE THREAD

**Issue by rpehlivan1** [chakra-ui/chakra-ui#4225]
# 🐛 Bug report

Upon using Drawer/Modal component and running Axe Devtools an error is thrown for **_tabIndex=1_**, stating _**"Elements should not have tabindex greater than zero"**_

## 💥 Steps to reproduce

1. Go to https://codesandbox.io/s/black-water-l8i6l?file=/src/index.js
2. Click on Open Modal/Open Drawer button
3. Open developer console
4. Install Axe Devtools chrome extension , run a scan with the drawer/modal open. https://chrome.google.com/webstore/detail/axe-devtools-web-accessib/lhdoppojpmngadmnindnejefpokejbdd
5. See error 

## 💻 Link to reproduction

CodeSandbox reproduction: https://codesandbox.io/s/black-water-l8i6l?file=/src/index.js

## 🧐 Expected behavior

According to A11Y guidelines tabindex attribute values should not be greater than 0, we shouldn't be seeing this error.

## 🧭 Possible Solution
I've come up with a **_workaround_** by disabling **_trapFocus_** and using https://github.com/focus-trap/focus-trap-react to control the trapfocus myself, this way no A11Y issues are thrown.

## 🌍 System information

| Software         | Version(s) |
| ---------------- | ---------- |
| Chakra UI        |  latest          |
| Browser          |    chrome        |
| Operating System |    macOS        |

## 📝 Additional information
I'm attaching Deque modal component as an example, as how they're handling trap focus, without tabIndex 1 attribute value 
https://dequeuniversity.com/library/aria/simple-dialog

**Comment by segunadebayo** [comment]
Based on my quick investigation, this is caused by how `react-focus-lock` implements the focus trap. We could explore other potential solutions to achieve the same but without this accessibility warning/issue.

**Comment by RoryMusinskas** [comment]
+1 for implementing a fix for this issue. I had to implement the same fix as @rpehlivan1 to work around axe unit testing failures.

Thanks @rpehlivan1 for documenting a workaround.

**Comment by rpehlivan1** [comment]
Up!

**Comment by goodwin64** [comment]
> workaround by disabling trapFocus  

Hi @rpehlivan1, could you pls specify how did you disable the `trapFocus`? I have the same issue but don't see such a prop on the `Modal` component.
